### PR TITLE
CORDA-1471 Database schema setup for internal tables via Liquibase

### PR DESCRIPTION
**Pull request by szymonsztuka**
Internal tables (the tables from node and finance modules) are now tracked /created by Liquibase script.
For other tables as before Hibernate is responsible for table creation.
The PR scope added Liquibase library, setup code `SchemaMigration` and XML scripts and from Enterprise.

For existing database installation - the node will auto-upgrade to use Liquibase.
Method `migrateOlderDatabaseToUseLiquibase ` checks for any 3.X existing Corda database to upgrade database to use Liquibase. When the existing database without Liquibase integral tables is detected, the node (at startup) will create Liquibase tracking tables and fill them with all migration scripts (marked as done), this ensure the database will look as it would use Liquibase from the beginning. 
The database changes gradually introduced by the subsequent post 3.0 releases (3.1, 3.2) are conditionally run by Liquibase, the change if check if it's not present already so the database can be migrated from any 3.X version.
So far the conditional migrations are: 
node-core.changelog-postgres-blob.xml
node-core.changelog-v8.xml
Table modifications in these 2 conditional migrations scripts replaced earlier validation checks done via manual JDBC queries (removed methods `checkCorrectAttachmentsContractsTableName` and `checkCorrectCheckpointTypeOnPostgres` from `CordaPersistence`)

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/SchemaMigration.kt [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+    /**
+     * Will run the Liquibase migration on the actual database.
+     */
+    private fun runMigration(existingCheckpoints: Boolean) = doRunMigration(run = true, outputWriter = null, check = false, existingCheckpoints = existingCheckpoints)
+
+    /**
+     * Ensures that the database is up to date with the latest migration changes.
+     */
+    private fun checkState() = doRunMigration(run = false, outputWriter = null, check = true)
```
Comment (tudor-malene) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/SchemaMigration.kt]: the outputWriter is not needed

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/SchemaMigration.kt [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    }
+
+    /**
+     * Main entry point to the schema migration.
+     * Called during node startup.
+     */
+    fun nodeStartup(existingCheckpoints: Boolean) {
+        when {
+            databaseConfig.initialiseSchema -> {
+                migrateOlderDatabaseToUseLiquibase()
```
Comment (tudor-malene) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/SchemaMigration.kt]: Check for existing checkpoints first before calling migrateOlderDatabaseToUseLiquibase

**Comment by szymonsztuka** [comment]
Comments addressed.

**Comment by szymonsztuka** [comment]
@distributedleetravis Since Tudor approved this PR I needed to port 2 additional Liquibase scripts to follow DDL schema changes (scripts are ported from Enterprise repo0, and I remove integration tests testing upgrade from over Corda version as this to much maintenance and needs to be tests in the other way outside integration tests  (the test tries to modify database to revert to previous Corda version in order to test upgrade).  
All additional changes from 24 Aug.

**Pull request merged**